internal/indexer/postgres/affected_manifests_e2e_test.go

```diff
@@ -0,0 +1,294 @@
+	"github.com/lib/pq"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/internal/indexer"
+	"github.com/quay/claircore/test/integration"
+	"github.com/quay/claircore/test/log"
+)
+
+type affectedE2E struct {
+	failed bool
+	store  indexer.Store
+	db     *sqlx.DB
+	ctx    context.Context
+	ir     claircore.IndexReport
+	vr     claircore.VulnerabilityReport
+}
+
+func TestAffectedE2E(t *testing.T) {
+	integration.Skip(t)
+	ctx, done := context.WithCancel(context.Background())
+	defer done()
+	ctx = log.TestLogger(ctx, t)
+	db, store, teardown := TestStore(ctx, t)
+	defer teardown()
+
+	table := []struct {
+		// name of the defined affectedE2E test
+		name string
+		// file name of index report in ./testdata
+		irFName string
+		// file name of vuln report in ./testdata
+		vrFName string
+	}{
+		// these fixtures have a subtly. these fixtures
+		// were generated against the same database
+		// to esure all ids are sequentially increasing.
+		//
+		// if you are to add fixtures here you must generate
+		// this current set + your new fixtures against the database
+		// to ensure no record id overlap.
+		{
+			name:    "amazonlinux 1",
+			irFName: "amazonlinux:1.index.json",
+			vrFName: "amazonlinux:1.report.json",
+		},
+		{
+			name:    "debian 8",
+			irFName: "debian:8.index.json",
+			vrFName: "debian:8.report.json",
+		},
+		{
+			name:    "debian 9",
+			irFName: "debian:9.index.json",
+			vrFName: "debian:9.report.json",
+		},
+		{
+			name:    "debian 10",
+			irFName: "debian:10.index.json",
+			vrFName: "debian:10.report.json",
+		},
+		{
+			name:    "ubi 8",
+			irFName: "ubi.index.json",
+			vrFName: "ubi.report.json",
+		},
+		{
+			name:    "ubuntu 16.04",
+			irFName: "ubuntu:16.04.index.json",
+			vrFName: "ubuntu:16.04.report.json",
+		},
+		{
+			name:    "ubuntu 18.04",
+			irFName: "ubuntu:18.04.index.json",
+			vrFName: "ubuntu:18.04.report.json",
+		},
+		{
+			name:    "ubuntu 19.10",
+			irFName: "ubuntu:19.10.index.json",
+			vrFName: "ubuntu:19.10.report.json",
+		},
+		{
+			name:    "ubuntu 20.04",
+			irFName: "ubuntu:20.04.index.json",
+			vrFName: "ubuntu:20.04.report.json",
+		},
+		{
+			name:    "mitmproxy 4.0.1",
+			irFName: "mitmproxy:4.0.1.index.json",
+			vrFName: "mitmproxy:4.0.1.report.json",
+		},
+	}
+
+	for _, tt := range table {
+		// grab and deserialize test data
+		irPath := filepath.Join("testdata", tt.irFName)
+		vrPath := filepath.Join("testdata", tt.vrFName)
+		irFD, err := os.Open(irPath)
+		if err != nil {
+			t.Fatalf("fd open for ir failed: %v", err)
+		}
+		vrFD, err := os.Open(vrPath)
+		if err != nil {
+			t.Fatalf("fd open for vr failed: %v", err)
+		}
+
+		var ir claircore.IndexReport
+		var vr claircore.VulnerabilityReport
+		{
+			err := json.NewDecoder(irFD).Decode(&ir)
+			if err != nil {
+				t.Fatalf("could not decode ir: %v", err)
+			}
+		}
+		{
+			err := json.NewDecoder(vrFD).Decode(&vr)
+			if err != nil {
+				t.Fatalf("could not decode vr: %v", err)
+			}
+		}
```

Comment:
these braces create a new lexical scope so the same block can be used inside, why? because I am lazy and like to copy and paste, and I don't like that a small change is necessary when err is defined in the same scope but you'd like to reuse the var.